client/my-sites/checkout/checkout/domain-details-form.jsx

```diff
@@ -115,14 +117,24 @@ export class DomainDetailsForm extends PureComponent {
 	componentDidUpdate( prevProps, prevState ) {
 		const previousFormValues = formState.getAllFieldValues( prevState.form );
 		const currentFormValues = formState.getAllFieldValues( this.state.form );
+
 		if ( ! isEqual( previousFormValues, currentFormValues ) ) {
 			this.props.updateContactDetailsCache( this.getMainFieldValues() );
 		}
 	}
 
 	loadFormStateFromRedux = fn => {
 		// only load the properties relevant to the main form fields
-		fn( null, pick( this.props.contactDetails, this.fieldNames ) );
+		const formStateFromRedux = pick( this.props.contactDetails, this.fieldNames );
+		const { geo } = this.props;
+		if ( ! formStateFromRedux.countryCode ) {
+			formStateFromRedux.countryCode = get( geo, 'country_short', '' );
+		}
+		if ( ! formStateFromRedux.city ) {
+			formStateFromRedux.city = get( geo, 'city', '' );
+		}
```

Comment:
Would it be less confusing for the user if we were to populate the city value _only_ if a countryCode does not exist? Something like:

```
		if ( ! formStateFromRedux.countryCode ) {
			formStateFromRedux.countryCode = get( geo, 'country_short', '' );
			formStateFromRedux.city = formStateFromRedux.city || get( geo, 'city', '' );
		}
```
This would avoid the situations where users complete the form with details that do not match their geo location. For example, say I'm on a glamorous vacation and decide to complete a purchase. I choose 'Italy' as my country, but leave the city blank. When I return to the form, the country is as I left it, however the city is now 'Hong Kong'. 🇭🇰

Reply:
Nice catch 👍 (please feel free to fix this if you want with the other modifications)